presto-raptor/src/main/java/com/facebook/presto/raptor/metadata/DatabaseShardManager.java

```diff
@@ -287,6 +289,14 @@ public void addColumn(long tableId, ColumnInfo column)
                 if (attempts >= MAX_ADD_COLUMN_ATTEMPTS) {
                     throw metadataError(e);
                 }
+                log.warn(e, "Failed to alter table with attempt %d, will retry. SQL: %s", attempts, sql);
```

Comment:
Use `%s` rather than `%d` unless you need specific formatting like `%02d`. It's safer since it works for any type.

Reply:
"with attempt" -> "on attempt"
```java
log.warn(e, "Failed to alter table on attempt %s, will retry. SQL: %s", attempts, sql);

Reply:
Done